airbrus.mt

```diff
@@ -68,6 +68,55 @@ def UTF8JSON :DeepFrozen := composeCodec(UTF8, JSON)
 def makeLineTube() as DeepFrozen:
     return makePumpTube(makeSplitPump(b`$\n`))
 
+def make_giftExchange() :Any as DeepFrozen:
+    def givers  := [].asMap().diverge()
+    def wanters := [].asMap().diverge()
+        
+    def give(giver :Str, acceptor :Str, giftName :Str, gift :Any) :Void :
+        def giftBox := gift  # Allways wrap your presents! (For when gifts are promises)
+        if (wanters.contains(acceptor):
+            def wants := wanters[acceptor]
+            if (wants.contains(giftName)):
+              wants.fetch(giftName)[1].resolve(&giftBox)
+              return
+        if (!givers.contains(giver)):
+            givers[giver] := [].asMap().diverge()
+        def gifts := givers[giver]
+        gifts[giftName] := &giftBox
+        return
```

Comment:
Not needed.

Reply:
Implict returns of control flow rub me the wrong way. Harmless to leave in.